### DISCUSSION TITLE
Water meter with 3 wire reed switch

### DISCUSSION THREAD

**Discussion by garudaonekh** [arendst/Tasmota#12856] [Q&A] [answered]
Hi,

I want to read pulse from this water meter. It has 3 PINs-blue is Normally Open and Black is Normally Close). The duration of close and open change according to the water flow rate.

I can use only normally open pin and use Debounce to count the pulse but it's not accurate when the flow is low because it'll keep connected for a long time which make it difficult to configure the debouncing period. 

Is there anyway to use all the 3 wire and signal each other to count the pulse correctly?
![photo_2021-08-10 13 53 06](https://user-images.githubusercontent.com/2074857/128825691-248f50b8-23ea-43ec-ad01-1a2bcd27670d.jpeg)

**Comment by sfromis** [comment, marked as answer]
Did you try configuring a counter pin, and use the CounterType command to set it as a "pulse timer"?

**Reply by garudaonekh** [reply]
Yes, and set the BounceTime to 50 but it increments incorrectly. I think because it keep connected very long time when the water flow is low.

**Reply by sfromis** [reply]
Not sure why you want to use BounceTime at all, if you have configured for "pulse timer" instead of count.

**Reply by garudaonekh** [reply]
Sorry, I am confused. I refer to counterdebounce. and set countertype 0

**Reply by sfromis** [reply]
Let me return to my first question.... Did you try configuring as "pulse timer" instead of "pulse counter", in the CounterType command?

**Reply by garudaonekh** [reply]
I set my PIN 26 to Pulse Counter and the following two command in the console.
counterdebounce 100
countertype 0
Now it works correctly. But I am not sure if it will keep correct count if the amount of water flow fluctuate. Let's see.
Thanks;

**Comment by Pewibe** [comment]
Another solution is to use simple hardware between water meter switch and your device input. See  https://www.mikrocontroller.net/articles/Entprellung#Wechselschalter or something similar as english page https://embeddedflakes.com/switch-debouncing/
